tutorials/run-spring-petclinic-on-app-engine-cloudsql.md

```diff
@@ -0,0 +1,131 @@
+    spring.datasource.password=my-smart-password
+```
+1. Update `pom.xml` to include the [Cloud SQL MySQL Socket Factory socket library](https://github.com/GoogleCloudPlatform/cloud-sql-mysql-socket-factory):
+```
+    <dependency>
+        <groupId>com.google.cloud.sql</groupId>
+        <artifactId>mysql-socket-factory</artifactId>
+        <version>1.0.2</version>
+    </dependency>
+```
+   The socket library allows you to connect to your Cloud SQL instance for local testing and deployment.
+1. Restart the Spring Boot application using the mysql [profile](http://docs.spring.io/spring-boot/docs/current/maven-plugin/examples/run-profiles.html):
+```
+   ./mvnw -Drun.profiles=mysql spring-boot:run
+```
+  Access the application homepage http://localhost:8080 via your web browser and add some data.
+1. You can verify the data exists in Cloud SQL by running queries agains the `petclinic` database using the [Cloud Shell](https://cloud.google.com/sql/docs/mysql/quickstart#connect_to_your_instance_using_the_db_client_client_in_the_cloud_shell).
+
+
+### Deploying to App Engine flexible environment on GCP
+Now that you've tested the application locally, you can deploy the application to the App Engine flexible environment.
+Once deployed, it will be accessible at https://<your-project-id>.appspot.com.
+
+1. App Engine flexible environment provides Maven plugins to make your build and deploy process extremely easy.
+   Add the following plugin to your `pom.xml`'s `build` plugins section:
+```
+  <plugin>
+    <groupId>com.google.cloud.tools</groupId>
+    <artifactId>appengine-maven-plugin</artifactId>
+    <version>1.3.1</version>
+  </plugin>
+```
+
+1. Create an `app.yaml` under `src/main/appengine` with the following contents.
+For more on configuring `app.yaml`, refer to [this resource](https://cloud.google.com/appengine/docs/flexible/java/configuring-your-app-with-app-yaml):
+```
+    runtime: java
+    env: flex
+
+    resources:
+      memory_gb: 2.3
+
+    handlers:
+    - url: /.*
+      script: this field is required, but ignored
+ ```
+
+1. App Engine flexible environment monitors the health of your application using the `/_ah/health` endpoint.
+(Note: A `200` or`404` status is interpreted as the application being healthy.)
+Because Spring Boot automatically provides a [health check endpoint](https://docs.spring.io/spring-boot/docs/current/reference/html/production-ready-endpoints.html#production-ready-health),
+we can hook that up as our health check endpoint.
+
+Update the following fields in `src/main/resources/application.properties`:
+```
+   management.contextPath=/_ah
+   spring.profiles.active=mysql
+```
+
+1. Run the following command to deploy your app:
+```
+    ./mvnw -DskipTests=true appengine:deploy
+```
+
+1. Visit `https://YOUR_PROJECT_ID.appspot.com` to access the Pet Clinic application, now running on GCP.
+View the application logs using the [Cloud Platform Console](https://console.cloud.google.com/logs/viewer).
```

Comment:
You should add a "Next steps" section and link to the App Engine flexible Java docs and maybe some other Spring on GCP resources.